From 62972ab62dd216ef95df90187487d14bc86e5def Mon Sep 17 00:00:00 2001
From: Peter Williams <peter@newton.cx>
Date: Mon, 3 Sep 2018 17:36:09 -0400
Subject: [PATCH 4/8] Fix quoting when invoking "windres"

Version strings were being quoted too much. I'm not sure why this wasn't a
problem elsewhere, but it's definitely a problem on Windows.
---
 gettext-runtime/intl/Makefile.am        | 2 +-
 gettext-runtime/libasprintf/Makefile.am | 2 +-
 gettext-runtime/src/Makefile.am         | 2 +-
 gettext-tools/libgettextpo/Makefile.am  | 2 +-
 gettext-tools/src/Makefile.am           | 2 +-
 libtextstyle/lib/Makefile.am            | 2 +-
 6 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/gettext-runtime/intl/Makefile.am b/gettext-runtime/intl/Makefile.am
index 2bed67b..ae4123e 100644
--- a/gettext-runtime/intl/Makefile.am
+++ b/gettext-runtime/intl/Makefile.am
@@ -411,7 +411,7 @@ libintl.res.lo: $(srcdir)/libintl.rc
 	sed_extract_minor='/^[0-9][0-9]*[.][0-9]/{'$${nl}'s/^[0-9]*[.]\([0-9]*\).*/\1/p'$${nl}q$${nl}'}'$${nl}'c\'$${nl}0$${nl}q; \
 	sed_extract_subminor='/^[0-9][0-9]*[.][0-9][0-9]*[.][0-9]/{'$${nl}'s/^[0-9]*[.][0-9]*[.]\([0-9]*\).*/\1/p'$${nl}q$${nl}'}'$${nl}'c\'$${nl}0$${nl}q; \
 	$(LIBTOOL) --tag=RC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(RC) \
-	  "-DPACKAGE_VERSION_STRING=\\\"$(VERSION)\\\"" \
+	  "-DPACKAGE_VERSION_STRING=\"$(VERSION)\"" \
 	  "-DPACKAGE_VERSION_MAJOR="`echo '$(VERSION)' | sed -n -e "$$sed_extract_major"` \
 	  "-DPACKAGE_VERSION_MINOR="`echo '$(VERSION)' | sed -n -e "$$sed_extract_minor"` \
 	  "-DPACKAGE_VERSION_SUBMINOR="`echo '$(VERSION)' | sed -n -e "$$sed_extract_subminor"` \
diff --git a/gettext-runtime/libasprintf/Makefile.am b/gettext-runtime/libasprintf/Makefile.am
index c0eb406..4ca6b0a 100644
--- a/gettext-runtime/libasprintf/Makefile.am
+++ b/gettext-runtime/libasprintf/Makefile.am
@@ -99,7 +99,7 @@ libasprintf.res.lo: $(srcdir)/libasprintf.rc
 	sed_extract_minor='/^[0-9][0-9]*[.][0-9]/{'$${nl}'s/^[0-9]*[.]\([0-9]*\).*/\1/p'$${nl}q$${nl}'}'$${nl}'c\'$${nl}0$${nl}q; \
 	sed_extract_subminor='/^[0-9][0-9]*[.][0-9][0-9]*[.][0-9]/{'$${nl}'s/^[0-9]*[.][0-9]*[.]\([0-9]*\).*/\1/p'$${nl}q$${nl}'}'$${nl}'c\'$${nl}0$${nl}q; \
 	$(LIBTOOL) --tag=RC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(RC) \
-	  "-DPACKAGE_VERSION_STRING=\\\"$(VERSION)\\\"" \
+	  "-DPACKAGE_VERSION_STRING=\"$(VERSION)\"" \
 	  "-DPACKAGE_VERSION_MAJOR="`echo '$(VERSION)' | sed -n -e "$$sed_extract_major"` \
 	  "-DPACKAGE_VERSION_MINOR="`echo '$(VERSION)' | sed -n -e "$$sed_extract_minor"` \
 	  "-DPACKAGE_VERSION_SUBMINOR="`echo '$(VERSION)' | sed -n -e "$$sed_extract_subminor"` \
diff --git a/gettext-runtime/src/Makefile.am b/gettext-runtime/src/Makefile.am
index e9974c4..9da1c7b 100644
--- a/gettext-runtime/src/Makefile.am
+++ b/gettext-runtime/src/Makefile.am
@@ -70,7 +70,7 @@ RELOCATABLE_STRIP = :
 if WOE32
 WOE32_LDADD = gettext.res
 gettext.res : $(top_srcdir)/../windows/gettext.rc
-	$(WINDRES) `$(SHELL) $(top_srcdir)/../windows/windres-options --escape $(VERSION)` -i $(top_srcdir)/../windows/gettext.rc -o gettext.res --output-format=coff
+	$(WINDRES) `$(SHELL) $(top_srcdir)/../windows/windres-options $(VERSION)` -i $(top_srcdir)/../windows/gettext.rc -o gettext.res --output-format=coff
 MOSTLYCLEANFILES += gettext.res
 else
 WOE32_LDADD =
diff --git a/gettext-tools/libgettextpo/Makefile.am b/gettext-tools/libgettextpo/Makefile.am
index 5fc6708..98d65cc 100644
--- a/gettext-tools/libgettextpo/Makefile.am
+++ b/gettext-tools/libgettextpo/Makefile.am
@@ -146,7 +146,7 @@ EXTRA_DIST += libgettextpo.rc
 if WOE32
 WOE32_LIBADD = libgettextpo.res.lo
 libgettextpo.res.lo : $(srcdir)/libgettextpo.rc
-	$(LIBTOOL) --tag=RC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(RC) `$(SHELL) $(top_srcdir)/../windows/windres-options --escape $(VERSION)` -i $(srcdir)/libgettextpo.rc -o libgettextpo.res.lo --output-format=coff
+	$(LIBTOOL) --tag=RC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(RC) `$(SHELL) $(top_srcdir)/../windows/windres-options $(VERSION)` -i $(srcdir)/libgettextpo.rc -o libgettextpo.res.lo --output-format=coff
 MOSTLYCLEANFILES += libgettextpo.res.lo
 else
 WOE32_LIBADD =
diff --git a/gettext-tools/src/Makefile.am b/gettext-tools/src/Makefile.am
index f77129d..0f77ee8 100644
--- a/gettext-tools/src/Makefile.am
+++ b/gettext-tools/src/Makefile.am
@@ -756,7 +756,7 @@ RELOCATABLE_STRIP = :
 if WOE32
 WOE32_LDADD = gettext.res
 gettext.res : $(top_srcdir)/../windows/gettext.rc
-	$(WINDRES) `$(SHELL) $(top_srcdir)/../windows/windres-options --escape $(VERSION)` -i $(top_srcdir)/../windows/gettext.rc -o gettext.res --output-format=coff
+	$(WINDRES) `$(SHELL) $(top_srcdir)/../windows/windres-options $(VERSION)` -i $(top_srcdir)/../windows/gettext.rc -o gettext.res --output-format=coff
 MOSTLYCLEANFILES += gettext.res
 else
 WOE32_LDADD =
diff --git a/libtextstyle/lib/Makefile.am b/libtextstyle/lib/Makefile.am
index bf4b408..59810c1 100644
--- a/libtextstyle/lib/Makefile.am
+++ b/libtextstyle/lib/Makefile.am
@@ -268,7 +268,7 @@ libtextstyle.res.lo: $(srcdir)/libtextstyle.rc
 	sed_extract_minor='/^[0-9][0-9]*[.][0-9]/{'$${nl}'s/^[0-9]*[.]\([0-9]*\).*/\1/p'$${nl}q$${nl}'}'$${nl}'c\'$${nl}0$${nl}q; \
 	sed_extract_subminor='/^[0-9][0-9]*[.][0-9][0-9]*[.][0-9]/{'$${nl}'s/^[0-9]*[.][0-9]*[.]\([0-9]*\).*/\1/p'$${nl}q$${nl}'}'$${nl}'c\'$${nl}0$${nl}q; \
 	$(LIBTOOL) --tag=RC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(RC) \
-	  "-DPACKAGE_VERSION_STRING=\\\"$(VERSION)\\\"" \
+	  "-DPACKAGE_VERSION_STRING=\"$(VERSION)\"" \
 	  "-DPACKAGE_VERSION_MAJOR="`echo '$(VERSION)' | sed -n -e "$$sed_extract_major"` \
 	  "-DPACKAGE_VERSION_MINOR="`echo '$(VERSION)' | sed -n -e "$$sed_extract_minor"` \
 	  "-DPACKAGE_VERSION_SUBMINOR="`echo '$(VERSION)' | sed -n -e "$$sed_extract_subminor"` \
